### PR TITLE
Initial scaffolding for Filter action

### PR DESCRIPTION
**Pull request by dannyroosevelt**

### CODE REVIEW

**Comment by dylburger** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:

- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible

**Comment by dannyroosevelt** [comment]
There’s a strange bug that’s happening, and I’m not sure if it’s in the component code or the UI, but we seem to be adding the `logicalOperator` prop when I change the `condition` prop config: https://capture.dropbox.com/sVBCyEuflxptkgWF cc @js07 can you try and take a look also?

**Comment by js07** [comment]
@dannyroosevelt I think that's a bug in the UI. Typically props added by `additionalProps` don't have a "Hide" button, even if they are `optional`. When `logicalOperator` has the property `reloadProps: true`, the UI seems to treat it as a static prop, making it hide-able.

**Comment by dannyroosevelt** [comment]
> @dannyroosevelt I think that's a bug in the UI. Typically props added by `additionalProps` don't have a "Hide" button, even if they are `optional`. When the `logicalOperator` prop has the property `reloadProps: true`, the UI seems to treat it as a static prop, making it hide-able.

Oh, so shouldn't the UI treat optional props that are added via `additionalProps`? Is that the bug to fix?

**Comment by js07** [comment]
I think it's intended that the additional props aren't currently hideable (but could still be "optional" / left blank). I think the bug to fix is being able to "hide" some additional props (`logicalOperator`) when an additional prop has `reloadProps: true`. If `logicalOperator` is not hidden, then it's not duplicated when props are reloaded.

Alternatively, the UI could allow optional props added via additional props to be hidden, and dedupe hidden props when props are reloaded.

**Comment by andrewjschuang** [comment]
> I think it's intended that the additional props aren't currently hideable (but could still be "optional" / left blank). I think the bug to fix is being able to "hide" some additional props (`logicalOperator`) when an additional prop has `reloadProps: true`. If `logicalOperator` is not hidden, then it's not duplicated when props are reloaded.
> 
> Alternatively, the UI could allow optional props added via additional props to be hidden, and dedupe hidden props when props are reloaded.

I changed for now the behaviour for the `logicalOperator` prop to non-optional, making it unable to be hidden. Also added a default value `""` so that it's not required to be filled

File: components/filter/filter.app.mjs [hunk 1]
```diff
@@ -1,11 +1,225 @@
...
+      //   input = this.convertToNumber(input);
+      //   return new Date(input);
+      // } catch (err) {
+      //   // convert a string
+      //   input = new Date(input);
+      //   if (input == "Invalid Date") {
+      //     throw new Error("Input can not be converted to datetime");
+      //   }
+      //   return input;
+      // }
```
Comment (michelle0927) [inline components/filter/filter.app.mjs:121]: Were these comments left in from testing?
```suggestion
```

**Review by michelle0927 — DISMISSED** [review]
This component will be very useful! There appear to be some stray comments, but I'm approving.

**Review by michelle0927 — APPROVED** [review]

**Pull request merged**